src/cloud-api-adaptor/pkg/securecomms/wnssh/kbsclient.go

```diff
@@ -0,0 +1,86 @@
+package wnssh
+
+import (
+	"bytes"
+	"crypto"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type KbsClient struct {
+	secretKey crypto.PrivateKey
+	url       string
+}
+
+func InitKbsClient(url string) *KbsClient {
+	return &KbsClient{
+		url: url,
+	}
+}
+
+func (kc *KbsClient) SetPemSecret(keyBytes []byte) error {
+	key, err := jwt.ParseEdPrivateKeyFromPEM(keyBytes)
+	if err != nil {
+		return fmt.Errorf("SetPemSecret unable to parse as private key: %w", err)
+	}
+	kc.secretKey = key
+	return nil
+}
+
+func (kc *KbsClient) addToken(req *http.Request) error {
+	now := time.Now()
+	//Ed25519KeyPair
+	token := jwt.NewWithClaims(jwt.SigningMethodEdDSA,
+		jwt.MapClaims{
+			"issued_at":      now.Unix(),
+			"expires_at":     now.Add(time.Hour * 24).Unix(),
+			"invalid_before": now.Unix(),
+			"audiences":      "",
+			"issuer":         "",
+			"jwt_id":         "",
+			"subject":        "",
+			"nonce":          "",
+			"custom":         "",
+		})
+
+	tokenString, err := token.SignedString(kc.secretKey)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", tokenString))
+	return nil
+}
+
+func (kc *KbsClient) PostResource(path string, data []byte) error {
+	url := fmt.Sprintf("%s/kbs/v0/resource/%s", kc.url, path)
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	if err != nil {
+		return fmt.Errorf("KbsClient faield to PostResource - %v", err)
+	}
+	err = kc.addToken(req)
+	if err != nil {
+		return fmt.Errorf("KbsClient failed to create token - %v", err)
```

Comment:
What does that mean when this happens? Is this user-actionable? Does this happens when the KbsClient actually rejects the connection (i.e. would the following rewording be better)?

```suggestion
		return fmt.Errorf("KbsClient token creation error: %v", err)
```

(That would be better if the error is along the lines of "access denied".

Reply:
The WN reads a  "kbs-client" secret the Kubernetes system. 
This secret should include a valid private key
If this secret is not valid and it will fail when trying to sign a token that is needed to communicate with the KBS.
Failing to sign the token will result in an error. 
(This is not an access denied case since the request was not yet sent to the KBS, it is a failure in the client prior to sending the request).

I will improve the error message.